### ISSUE TITLE
`implicit_reopen` does not report on class type aliases

### ISSUE THREAD

**Issue by srawlins** [dart-lang/linter#4236]
**Describe the issue**

I would expect `implicit_reopen` to be reported with this code:

```dart
final class C {}

mixin M {}

base class D = C with M;
```

The `D` class _does_ re-open `C`, right? I'm not 100% sure 😅 

CC @kallentu @pq @munificent

**Comment by eernstg** [comment]
Yes, `D` does re-open `C`. The intuition is that `D` can inherit member implementations from `C`, and `C` has a modifier `final` that implies `interface` ("`final == interface && base`"), and hence `D` should also be at least `interface` in order to avoid re-opening.
